File: python/7 kyu/434.py
# https://www.codewars.com/kata/63b4758f27f8e5000fc1e427/train/python

# Passed

def yahtzee_upper(dice):
    value_to_scores_dict = dict()
    for value in dice:
        if value not in value_to_scores_dict:
            value_to_scores_dict[value] = 0
            
        value_to_scores_dict[value] += value
        
    result = max(value_to_scores_dict.values())
    return result

output = yahtzee_upper([15, 9, 9, 8, 9])
print(output)

File: nodejs/6 kyu/434.js
// https://www.codewars.com/kata/5503013e34137eeeaa001648/train/javascript

// Passed

function diamond(n) {
  if (n <= 0 || n % 2 === 0) {
    return null;
  }

  let upsideDownTriangleRows = [];
  let createRow = (size, rowSize) => {
    let whitespaceCount = (rowSize - size) / 2;
    let row = " ".repeat(whitespaceCount) + "*".repeat(size);
    return row;
  };
  for (let asterixCount = n; asterixCount >= 1; asterixCount -= 2) {
    let row = createRow(asterixCount, n);
    upsideDownTriangleRows.push(row);
  }

  let diamondRows = [...upsideDownTriangleRows]
    .reverse()
    .concat(upsideDownTriangleRows.slice(1, upsideDownTriangleRows.length + 1));
  let result = diamondRows.join("\n") + "\n";
  return result;
}

const output = diamond(5);
console.log(output);